src/tests/system/tests/test_sssctl.py

```diff
@@ -242,6 +180,55 @@ def test_sssctl__check_missing_domain_name(client: Client):
     assert ex.match(r"Section \[domain\/\] is not allowed. Check for typos.*"), "Wrong error message was returned"
 
 
+@pytest.mark.parametrize(
+    "pattern,repl,expected",
+    [
+        ("id_provider = ", "id_provider ", ("Equal sign is missing", "stderr")),
+        pytest.param(
+            "id_provider.*",
+            "id_provider = invalid",
+            ("Attribute 'id_provider' in section 'domain/local' has an invalid value: invalid", "stdout"),
+            marks=[pytest.mark.ticket(bz=2100789), pytest.mark.importance("high")],
+        ),
+        pytest.param(
+            "id_provider.*",
+            "",
+            ("Attribute 'id_provider' is missing in section 'domain/local'.", "stdout"),
+            marks=[pytest.mark.ticket(bz=2100789), pytest.mark.importance("high")],
+        ),
+        ("id_provider", "id_@provider", ("Attribute 'id_@provider' is not allowed in section", "stdout")),
+        ("domain/", "d$main/", ("Section [d$main/local] is not allowed", "stdout")),
+        ("domain/local", "domain/local@", ("Section [domain/local@] is not allowed", "stdout")),
+        ("domain/local", "domainlocal", ("Section [domainlocal] is not allowed", "stdout")),
+        (".sssd.", "[sssdx]", ("Section [sssdx] is not allowed", "stdout")),
+        (".nss.", "[nssx", ("No closing bracket", "stderr")),
+        (".domain/local.", "domain/local]", ("Equal sign is missing", "stderr")),
+    ],
+)
+@pytest.mark.tools
+@pytest.mark.topology(KnownTopology.Client)
+def test_sssctl__check_invalid_section_name(client: Client, pattern: str, repl: str, expected: tuple[str, str]):
+    """
+    :title: sssctl prints appropriate error message with invalid configurations
+    :setup:
+        1. Start SSSD
+        2. Edit sssd.conf with invalid value
+    :steps:
+        1. Validate configuration file using sssctl config-check
+    :expectedresults:
+        1. sssctl configuration check fails with the correct output
+    :customerscenario: False
+    """
+    client.sssd.common.local()
+    client.sssd.start()
+    conf = re.sub(pattern, repl, client.fs.read("/etc/sssd/sssd.conf"))
+    client.fs.write("/etc/sssd/sssd.conf", conf, mode="600")
+
+    result = client.sssctl.config_check()
+    assert result.rc != 0, "Config-check did not detect misconfigured config!"
+    assert expected[0] in getattr(result, expected[1]), "Wrong error message on stderr!"
```

Comment:
Well, hard coding "Wrong error message on **stderr**!" is wrong, in should be "expected[1]".

But I tend to agree with @jakub-vavra-cz here - tests gets a bit unreadable.

IIRC, errors reported on 'stderr' are parsing errors, where SSSD wouldn't event start.
And errors resported on 'stdout' are semantic issue, that are reported but ignored by SSSD.
Maybe it makes sense to keep at least 2 tests for those 2 very different issue types (syntax vs semantic).